### DISCUSSION TITLE
Running iLEAPP on macOS, no module named 'magic' found

### DISCUSSION THREAD

**Discussion by WKirklandJr** [abrignoni/iLEAPP#622] [Q&A] [answered]
Hello,

I was following the install instructions here (https://www.youtube.com/watch?v=fEYV5vVAdu4) and encountered an error when I tried installing the dependendencies from requirements.txt. For some reason it is unable to find magic & python magic. 

I have XCode installed however I am on a MacBook Pro Apple M2 Max Chip. 
<img width="913" alt="Screenshot 2023-11-29 at 1 13 13 PM" src="https://github.com/abrignoni/iLEAPP/assets/73005151/6d094795-6253-4b09-a82c-b15eded9930d">
<img width="640" alt="Screenshot 2023-11-29 at 1 11 22 PM" src="https://github.com/abrignoni/iLEAPP/assets/73005151/6fb8499a-ca5c-4ec2-9064-04839392071c">

**Comment by stark4n6** [comment]
I'm not sure if applicable on Mac but I know this issue gets fixed after installed the Visual Studio items on Windows https://youtu.be/7qvVFfBM2NU?si=B93oHWTmLaVPHN2h&t=116

**Comment by Johann-PLW** [comment]
I also had the same issue on a Mac Studio with an Apple M1 Max chip with a fresh install of Sonoma.
I don't know if it is related to the Python version or Xcode15 Command Line Tools as I don't have the same issue on a Mac Mini M1.
I presume this is not the best solution but I fixed that using [brew](https://brew.sh) and installing libmagic (brew install lib magic)
Don't forget to comment 'python-magic-bin==0.4.14; platform_system == "Darwin"' in requirements.txt files and type again pip install -r requirements.txt.
Hope it will help you.

**Reply by WKirklandJr** [reply]
Okay that helped a bit, I installed libmagic with brew and ran the pip3 install -r requirements.txt with no errors or issues after taking/commenting out the "python-magic-bin" however when I ran the iLEAPPGUI.py file, I got this:
<img width="656" alt="Screenshot 2023-11-29 at 6 03 38 PM" src="https://github.com/abrignoni/iLEAPP/assets/73005151/2310fb72-c7d0-43c3-871e-f308ea71d5ba">
Is it due to my lib magic being in the wrong directory of where it's looking?

**Comment by Johann-PLW** [comment, marked as answer]
I did a test in a VM with a fresh install of macOS Sonoma 14.1.1. and it worked.
I downloaded Python 3.11.6, opened a terminal and typed git --version to install Xcode Command Line tools, cloned the iLEAPP repo in my Documents folder and comment the 'python-magic-bin==0.4.14; platform_system == "Darwin"' line in the requirements.txt file.
In the terminal, I installed the dependencies (pip3 install -r requirements.txt) ant tried to launch ileappGUI.py but the libmagic module was not found.

I installed brew then the libmagic formula without any additional parameter (I.E: path)
<img width="552" alt="image" src="https://github.com/abrignoni/iLEAPP/assets/27498476/15386a20-f815-4b3e-9ea0-2a634b730556">

After that, I was able to start iLEAPP in both GUI and CLI.

Verify that the libmagic formula is available with 'brew list' command: 
<img width="552" alt="image" src="https://github.com/abrignoni/iLEAPP/assets/27498476/9f0919ee-efe4-4d1f-8018-97623e31c87f">
